course-outline.md

```diff
@@ -0,0 +1,105 @@
+
+## 3. Git and GitHub intro
+
+1. Know of the importance of version control: what it is and why we use it
+2. Use git’s basic functions: clone, add, commit, pull, and push
+3. Create a new repository on GitHub via the web interface
+
+## 4. Getting groovy with Git and GitHub
+
+1. View your Git history via git log and restore an older version of a file via git restore
+2. Deal with merge conflicts at the command line and in VS Code
+3. Know when you can stash your changes with git stash before pulling from GitHub
+4. Telling git to ignore certain files
+    - [521 lec 7 end](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/7-ssh-filenames-project-organization.html#tell-git-to-ignore-irrelevant-files-using-a-gitignore-file)
+
+## 5. Branches, forks, and streams… Welcome to the Git nature walk!
+
+1. Develop a feature on a branch in a Git repo.
+2. Merge the feature branch with the main branch.
+3. Create a fork of an existing GitHub repo.
+4. Open a pull request from your forked repo to the upstream repo.
+5. Sync your local and origin remote with an upstream remote.
+    - This is much easier now with the new GitHub interface so will not take much time
+6. Use git reset to move a branch back in the commit history.
+    - This might be too much in a single lecture, but there is no natural break earlier
+    - Additional resources for all git chapters:
+        - https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/1-lecture1-MDStools-shell-git-github.html#introduction-to-local-and-remote-version-control-software
+        - https://datasciencebook.ca/Getting-started-with-version-control.html
+        - https://merely-useful.tech/py-rse/git-cmdline.html
+
+## 6. Project organization and environments
+
+1. Organize projects and name files in a sound manner.
+    - [521 lec 7](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/7-ssh-filenames-project-organization.html#filenames-best-practices)
+2. Use virtual conda environments for Python.
+    - [521 lec 8](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/8-asking-effectively-and-virtual-environments.html#virtual-environments)
+
+## 7. JupyterLab and reproducible documents
+
+1. Be comfortable using JupyterLab
+    - [The first assignment in the programming course](https://github.ubc.ca/ubc-mds/prog-python-data-science-instructor/blob/master/source/assignment1/assignment1.ipynb) includes the Jupyter intro from lab 0 in 521.
+        - But I still think it is fine if we repeat some things and have an easy first assignment
+2. Create reproducible literate Python (e.g., reports and presentations) using Jupyter
```

Comment:
Presentation as separate point and spend more time on it? Maybe even deploying slides on GH pages? Then we could talk Pages in general here, which would make this chapter bigger.

Reply:
Otherwise GH pages in Ch8